src/lib.rs

```diff
@@ -1026,8 +1020,46 @@ pub struct PublishOptions<'a> {
     pub dry_run: bool,
 }
 
-fn add_registry_metadata(_package: &Package, bytes: &[u8], _path: &Path) -> Result<Vec<u8>> {
-    Ok(bytes.to_owned())
+/// Read metadata from `Cargo.toml` and add it to the component
+fn add_component_metadata(package: &Package, wasm: &[u8]) -> Result<Vec<u8>> {
+    let metadata = wasm_metadata::AddMetadata {
+        name: Some(package.name.clone()),
+        language: vec![("Rust".to_string(), "".to_string())],
+        processed_by: vec![(
+            env!("CARGO_PKG_NAME").to_string(),
+            option_env!("CARGO_VERSION_INFO")
+                .unwrap_or(env!("CARGO_PKG_VERSION"))
+                .to_string(),
+        )],
+        sdk: vec![],
+        authors: match package.authors.len() {
+            0 => None,
+            _ => Some(wasm_metadata::Authors::new(package.authors.join(","))),
+        },
+        description: package
+            .description
+            .as_ref()
+            .map(|d| wasm_metadata::Description::new(d.clone())),
+        licenses: package
+            .license
+            .as_ref()
+            .map(|s| wasm_metadata::Licenses::new(&s))
+            .transpose()?,
+        source: package
+            .repository
+            .as_ref()
+            .map(|s| wasm_metadata::Source::new(s.to_string().as_str()))
+            .transpose()?,
+        homepage: package
+            .homepage
+            .as_ref()
+            .map(|s| wasm_metadata::Homepage::new(s.to_string().as_str()))
+            .transpose()?,
+        // TODO: get the git commit hash
```

Comment:
Lets file an issue to resolve this soon?

Reply:
Done! https://github.com/bytecodealliance/cargo-component/issues/378